code/modules/surgery/bodyparts/head_hair_and_lips.dm

```diff
@@ -99,15 +102,24 @@
 			var/facial_hair_gradient_style = gradient_styles[GRADIENT_FACIAL_HAIR_KEY]
 			if(facial_hair_gradient_style != "None")
 				var/facial_hair_gradient_color = gradient_colors[GRADIENT_FACIAL_HAIR_KEY]
-				var/image/facial_hair_gradient_overlay = get_gradient_overlay(sprite_accessory.icon, sprite_accessory.icon_state, -HAIR_LAYER, SSaccessories.facial_hair_gradients_list[facial_hair_gradient_style], facial_hair_gradient_color, image_dir)
+				var/image/facial_hair_gradient_overlay = get_gradient_overlay(icon(sprite_accessory.icon, sprite_accessory.icon_state), -HAIR_LAYER, SSaccessories.facial_hair_gradients_list[facial_hair_gradient_style], facial_hair_gradient_color, image_dir)
 				. += facial_hair_gradient_overlay
 
 	var/image/hair_overlay
 	if(!(show_debrained && (head_flags & HEAD_DEBRAIN)) && !hair_hidden && hairstyle && (head_flags & HEAD_HAIR))
 		var/datum/sprite_accessory/hair/hair_sprite_accessory = SSaccessories.hairstyles_list[hairstyle]
 		if(hair_sprite_accessory)
+			var/icon/base_icon
+			if(LAZYLEN(hair_masks))
+				base_icon = icon(hair_sprite_accessory.icon, hair_sprite_accessory.icon_state)
+				for(var/mask in hair_masks)
+					var/icon/blend_with = icon('icons/mob/human/hair_masks.dmi', mask)
+					blend_with.Shift(SOUTH, hair_sprite_accessory.y_offset)
+					base_icon.Blend(blend_with, ICON_ADD)
```

Comment:
how expensive is this? could it be cached?

Reply:
This is about as expensive as the blend operation for hair gradients. It does not have a global cache per mask and hairstyle, but it does use the limb icon cache in [carbon_update_icons.dm](https://github.com/tgstation/tgstation/pull/87263/files#diff-0334e1f70c7955e63905abc4925ecc2a3e97f33a2164c3f2692f367f93609322). Updating that cache system to be global and more granular is something I looked at but is definitely out of scope for this PR.

Reply:
that might be enough, worth keeping an eye on cost in this proc though hair gradients are already gonna hit us worse anyhow. I am uncomfortable with overusing blends but what can you do.